config/dashboards/dashboards_config.go

```diff
@@ -128,6 +130,18 @@ func GetBuiltInMonitoringDashboards() MonitoringDashboardsList {
 	}
 }
 
+func GetNamespaceMonitoringDashboards(annotations map[string]string) MonitoringDashboardsList {
+	if dashboardYaml, ok := annotations[DashboardTemplateAnnotation]; ok {
+		if v, err := unmarshal(dashboardYaml); err == nil {
+			return MonitoringDashboardsList(v)
+		} else {
+			log.Errorf("Failed to unmarshall namespace dashboard yaml. err=%v", err)
```

Comment:
Is there a way we can get the namespace name that owns these annotations so we can add it to this error message? Or can we bubble up the err so the caller (which I assume knows what namespace these annotations came from) can log the name of the namespace along with this error message?

Think about what will happen if the user put invalid yaml in the annotation (which is very likely, given the complexity of the dashboard yaml). When this happens, the user will see the log message about the bad yaml, but the user will not be told WHICH namespace has the bad annotation. Without knowing the namespace, it could be difficult to find the bad yaml in order to fix it (especially if they have hundreds (or more) of namespaces).